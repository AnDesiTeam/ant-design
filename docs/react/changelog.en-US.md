---
order: 10
title: Change Log
toc: false
---

<embed src="../../CHANGELOG.en-US.md"></embed>
