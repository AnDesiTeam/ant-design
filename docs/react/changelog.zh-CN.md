---
order: 10
title: 更新日志
toc: false
---

<embed src="../../CHANGELOG.zh-CN.md"></embed>
