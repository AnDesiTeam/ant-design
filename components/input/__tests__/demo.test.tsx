import * as React from 'react';
import Input from '..';
import demoTest, { rootPropsTest } from '../../../tests/shared/demoTest';

demoTest('input');

rootPropsTest('input', (props) => <Input addonBefore="Bamboo" addonAfter="Light" {...props} />, {
  name: 'addon',
});

rootPropsTest('input', (props) => <Input.Password {...props} />, {
  name: 'password',
});
