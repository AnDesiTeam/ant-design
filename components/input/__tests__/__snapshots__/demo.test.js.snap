// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/input/demo/addon.md correctly 1`] = `
<div>
  <div
    style="margin-bottom:16px"
  >
    <span
      class="ant-input-group-wrapper"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <span
          class="ant-input-group-addon"
        >
          Http://
        </span>
        <input
          class="ant-input"
          type="text"
          value="mysite"
        />
        <span
          class="ant-input-group-addon"
        >
          .com
        </span>
      </span>
    </span>
  </div>
  <div
    style="margin-bottom:16px"
  >
    <span
      class="ant-input-group-wrapper"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <span
          class="ant-input-group-addon"
        >
          <div
            class="ant-select ant-select-enabled"
            style="width:90px"
          >
            <div
              aria-autocomplete="list"
              aria-controls=""
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection-selected-value"
                  style="display:block;opacity:1"
                  title="Http://"
                >
                  Http://
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select:none;-webkit-user-select:none"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-arrow-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    style="vertical-align:-0.125em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </span>
        <input
          class="ant-input"
          type="text"
          value="mysite"
        />
        <span
          class="ant-input-group-addon"
        >
          <div
            class="ant-select ant-select-enabled"
            style="width:80px"
          >
            <div
              aria-autocomplete="list"
              aria-controls=""
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection-selected-value"
                  style="display:block;opacity:1"
                  title=".com"
                >
                  .com
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select:none;-webkit-user-select:none"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-arrow-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    style="vertical-align:-0.125em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </span>
      </span>
    </span>
  </div>
  <div
    style="margin-bottom:16px"
  >
    <span
      class="ant-input-group-wrapper"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <input
          class="ant-input"
          type="text"
          value="mysite"
        />
        <span
          class="ant-input-group-addon"
        >
          <span
            class="anticon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              fill="currentColor"
              focusable="false"
              height="1em"
              width="1em"
            >
              <use
                href="#icon-setting-o"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`renders ./components/input/demo/allowClear.md correctly 1`] = `
<span
  class="ant-input-affix-wrapper"
>
  <input
    class="ant-input"
    placeholder="input with clear icon"
    type="text"
    value=""
  />
  <span
    class="ant-input-suffix"
  />
</span>
`;

exports[`renders ./components/input/demo/autosize-textarea.md correctly 1`] = `
<div>
  <textarea
    class="ant-input"
    placeholder="Autosize height based on content lines"
  />
  <div
    style="margin:24px 0"
  />
  <textarea
    class="ant-input"
    placeholder="Autosize height with minimum and maximum number of lines"
  />
</div>
`;

exports[`renders ./components/input/demo/basic.md correctly 1`] = `
<input
  class="ant-input"
  placeholder="Basic usage"
  type="text"
  value=""
/>
`;

exports[`renders ./components/input/demo/group.md correctly 1`] = `
<div>
  <span
    class="ant-input-group ant-input-group-lg"
  >
    <div
      class="ant-row"
      style="margin-left:-4px;margin-right:-4px"
    >
      <div
        class="ant-col ant-col-5"
        style="padding-left:4px;padding-right:4px"
      >
        <input
          class="ant-input"
          type="text"
          value="0571"
        />
      </div>
      <div
        class="ant-col ant-col-8"
        style="padding-left:4px;padding-right:4px"
      >
        <input
          class="ant-input"
          type="text"
          value="26888888"
        />
      </div>
    </div>
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <input
      class="ant-input"
      style="width:20%"
      type="text"
      value="0571"
    />
    <input
      class="ant-input"
      style="width:30%"
      type="text"
      value="26888888"
    />
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Zhejiang"
          >
            Zhejiang
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <input
      class="ant-input"
      style="width:50%"
      type="text"
      value="Xihu District, Hangzhou"
    />
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Option1"
          >
            Option1
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <input
      class="ant-input"
      style="width:50%"
      type="text"
      value="input content"
    />
    <div
      class="ant-input-number"
    >
      <div
        class="ant-input-number-handler-wrap"
      >
        <span
          aria-disabled="false"
          aria-label="Increase Value"
          class="ant-input-number-handler ant-input-number-handler-up "
          role="button"
          unselectable="unselectable"
        >
          <span
            aria-label="up"
            class="anticon anticon-up ant-input-number-handler-up-inner"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="up"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
              />
            </svg>
          </span>
        </span>
        <span
          aria-disabled="false"
          aria-label="Decrease Value"
          class="ant-input-number-handler ant-input-number-handler-down "
          role="button"
          unselectable="unselectable"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-input-number-handler-down-inner"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        aria-valuemin="-9007199254740991"
        class="ant-input-number-input-wrap"
        role="spinbutton"
      >
        <input
          autocomplete="off"
          class="ant-input-number-input"
          min="-9007199254740991"
          step="1"
          value=""
        />
      </div>
    </div>
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <input
      class="ant-input"
      style="width:50%"
      type="text"
      value="input content"
    />
    <span
      class="ant-calendar-picker"
      style="width:50%"
    >
      <div>
        <input
          class="ant-calendar-picker-input ant-input"
          placeholder="Select date"
          readonly=""
          value=""
        />
        <span
          aria-label="calendar"
          class="anticon anticon-calendar ant-calendar-picker-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="vertical-align:-0.125em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </div>
    </span>
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Option1-1"
          >
            Option1-1
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Option2-2"
          >
            Option2-2
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Between"
          >
            Between
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <input
      class="ant-input"
      placeholder="Minimum"
      style="width:100px;text-align:center"
      type="text"
      value=""
    />
    <input
      class="ant-input ant-input-disabled"
      disabled=""
      placeholder="~"
      style="width:30px;border-left:0;pointer-events:none;background-color:#fff"
      type="text"
      value=""
    />
    <input
      class="ant-input"
      placeholder="Maximum"
      style="width:100px;text-align:center;border-left:0"
      type="text"
      value=""
    />
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Sign Up"
          >
            Sign Up
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled"
      style="width:200px"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection__placeholder"
            style="display:block;user-select:none;-webkit-user-select:none"
            unselectable="on"
          >
            Email
          </div>
          <ul>
            <li
              class="ant-select-search ant-select-search--inline"
            >
              <div
                class="ant-select-search__field__wrap"
              >
                <input
                  class="ant-input ant-select-search__field"
                  type="text"
                  value=""
                />
                <span
                  class="ant-select-search__field__mirror"
                >
                   
                </span>
              </div>
            </li>
          </ul>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </span>
  <br />
  <span
    class="ant-input-group ant-input-group-compact"
  >
    <div
      class="ant-select ant-select-enabled"
      style="width:30%"
    >
      <div
        aria-autocomplete="list"
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
            ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1"
            title="Home"
          >
            Home
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-arrow-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
    <span
      class="ant-cascader-picker"
      style="width:70%"
      tabindex="0"
    >
      <span
        class="ant-cascader-picker-label"
      />
      <input
        autocomplete="off"
        class="ant-input ant-cascader-input "
        placeholder="Select Address"
        readonly=""
        tabindex="-1"
        type="text"
        value=""
      />
      <span
        aria-label="down"
        class="anticon anticon-down ant-cascader-picker-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          style="vertical-align:-0.125em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;

exports[`renders ./components/input/demo/password-input.md correctly 1`] = `
<span
  class="ant-input-password ant-input-affix-wrapper"
>
  <input
    action="click"
    class="ant-input"
    placeholder="input password"
    type="password"
    value=""
  />
  <span
    class="ant-input-suffix"
  >
    <span
      aria-label="eye-invisible"
      class="anticon anticon-eye-invisible ant-input-password-icon"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="eye-invisible"
        fill="currentColor"
        focusable="false"
        height="1em"
        style="vertical-align:-0.125em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
        />
        <path
          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`renders ./components/input/demo/presuffix.md correctly 1`] = `
<div>
  <span
    class="ant-input-affix-wrapper"
  >
    <span
      class="ant-input-prefix"
    >
      <span
        class="anticon"
        role="img"
        style="color:rgba(0,0,0,.25)"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-user-o"
          />
        </svg>
      </span>
    </span>
    <input
      class="ant-input"
      placeholder="Enter your username"
      type="text"
      value=""
    />
    <span
      class="ant-input-suffix"
    >
      <span
        class="anticon"
        role="img"
        style="color:rgba(0,0,0,.45)"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          <use
            href="#icon-info-circle-o"
          />
        </svg>
      </span>
    </span>
  </span>
  <br />
  <br />
  <span
    class="ant-input-affix-wrapper"
  >
    <span
      class="ant-input-prefix"
    >
      ￥
    </span>
    <input
      class="ant-input"
      type="text"
      value=""
    />
    <span
      class="ant-input-suffix"
    >
      RMB
    </span>
  </span>
</div>
`;

exports[`renders ./components/input/demo/search-input.md correctly 1`] = `
<div>
  <span
    class="ant-input-search ant-input-affix-wrapper"
    style="width:200px"
  >
    <input
      class="ant-input"
      placeholder="input search text"
      type="text"
      value=""
    />
    <span
      class="ant-input-suffix"
    >
      <span
        aria-label="search"
        class="anticon anticon-search ant-input-search-icon"
        role="img"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="1em"
          style="vertical-align:-0.125em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </span>
    </span>
  </span>
  <br />
  <br />
  <span
    class="ant-input-search ant-input-search-enter-button ant-input-group-wrapper"
  >
    <span
      class="ant-input-wrapper ant-input-group"
    >
      <input
        class="ant-input"
        placeholder="input search text"
        type="text"
        value=""
      />
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn ant-input-search-button ant-btn-primary"
          type="button"
        >
          <span
            aria-label="search"
            class="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              style="vertical-align:-0.125em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </span>
  <br />
  <br />
  <span
    class="ant-input-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
  >
    <span
      class="ant-input-wrapper ant-input-group"
    >
      <input
        class="ant-input ant-input-lg"
        placeholder="input search text"
        type="text"
        value=""
      />
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
          type="button"
        >
          <span>
            Search
          </span>
        </button>
      </span>
    </span>
  </span>
</div>
`;

exports[`renders ./components/input/demo/size.md correctly 1`] = `
<div
  class="example-input"
>
  <input
    class="ant-input ant-input-lg"
    placeholder="large size"
    type="text"
    value=""
  />
  <input
    class="ant-input"
    placeholder="default size"
    type="text"
    value=""
  />
  <input
    class="ant-input ant-input-sm"
    placeholder="small size"
    type="text"
    value=""
  />
</div>
`;

exports[`renders ./components/input/demo/textarea.md correctly 1`] = `
<textarea
  class="ant-input"
  rows="4"
/>
`;

exports[`renders ./components/input/demo/textarea-resize.md correctly 1`] = `
<div>
  <button
    class="ant-btn"
    style="margin-bottom:16px"
    type="button"
  >
    <span>
      Auto Resize: false
    </span>
  </button>
  <textarea
    class="ant-input"
    rows="4"
  >
    autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。autosize 属性适用于 textarea 节点，并且只有高度会自动变化。另外 autosize 可以设定为一个对象，指定最小行数和最大行数。ending
  </textarea>
</div>
`;

exports[`renders ./components/input/demo/tooltip.md correctly 1`] = `
<input
  class="ant-input"
  maxlength="25"
  placeholder="Input a number"
  style="width:120px"
  type="text"
  value=""
/>
`;
